### PR TITLE
Implement monthly and lifetime remaining fields in merch item

### LINKED ISSUE #194: Add 'monthlyRemaining', 'lifetimeRemaining' fields to PublicMerchItem response
Title. This is so that Item page UI can indicate to users that they can only buy `x` more of an item for the month, or for their lifetime

### PR DESCRIPTION
**Pull request by shravanhariharan2**
Closes #194 and #193

### CODE REVIEW

**Comment by shravanhariharan2** [comment]
Logic and tests are mostly done, just stuck on a bug where the TypeORM join for getting the item order counts is not working in the repository layer. Will look into this tomorrow

File: tests/merchStore.test.ts [hunk 1]
```diff
@@ -68,6 +68,98 @@ describe('archived merch collections', () => {
...
+    const unorderedItemParams = { uuid: unorderedItem.uuid };
+    const getUnorderedItemResponse = await merchStoreController.getOneMerchItem(unorderedItemParams, member);
+    const unchangedItem = getUnorderedItemResponse.item;
+
+    // make sure the un-ordered item's remaining counts didn't change
+    expect(unchangedItem.monthlyRemaining).toEqual(5);
+    expect(unchangedItem.lifetimeRemaining).toEqual(10);
+  });
+
+  test('monthly and lifetime remaining values for one item don\'t change if another item is ordered', async () => {
```
Comment (michl1001) [inline tests/merchStore.test.ts:123]: Redundant test

**Review by michl1001 — COMMENTED** [review]
Looks good, just need to remove a redundant test for the merch store

**Review by michl1001 — APPROVED** [review]
Looks good

**Pull request merged**